src/profile.c

```diff
@@ -365,88 +365,113 @@ static inline Int8 getTicks(void)
+            newticks = getTicks();
+
+            ticks = newticks - profileState.lastOutputtedTime;
+
+            // Basic sanity check
+            if (ticks < 0)
+                ticks = 0;
+            if ((profileState.minimumProfileTick == 0) ||
+                (ticks > profileState.minimumProfileTick) || (!visited)) {
+                int ticksDone;
+                if (profileState.minimumProfileTick == 0) {
+                    ticksDone = ticks;
+                }
+                else {
+                    ticksDone = (ticks / profileState.minimumProfileTick) *
+                                profileState.minimumProfileTick;
+                }
+                ticks -= ticksDone;
+                outputFilenameIdIfRequired(nameid);
+                fprintf(profileState.Stream,
+                        "{\"Type\":\"%c\",\"Ticks\":%d,\"Line\":%d,"
+                        "\"FileId\":%d}\n",
+                        exec ? 'E' : 'R', ticksDone, (int)line, (int)nameid);
+                profileState.lastOutputtedTime = newticks;
+                profileState.lastNotOutputted.line = -1;
+                profileState.lastOutputted.line = line;
+                profileState.lastOutputted.fileID = nameid;
+                profileState.lastOutputtedExec = exec;
+            }
+            else {
+                profileState.lastNotOutputted.line = line;
+                profileState.lastNotOutputted.fileID = nameid;
+            }
+        }
+        else {
+            outputFilenameIdIfRequired(nameid);
+            fprintf(profileState.Stream,
+                    "{\"Type\":\"%c\",\"Line\":%d,\"FileId\":%d}\n",
+                    exec ? 'E' : 'R', (int)line, (int)nameid);
+            profileState.lastOutputted.line = line;
+            profileState.lastOutputted.fileID = nameid;
+            profileState.lastOutputtedExec = exec;
+            profileState.lastNotOutputted.line = -1;
+        }
+    }
+}
+
 // exec : are we executing this statement
 // visit: Was this statement previously visited (that is, executed)
 static inline void outputStat(Stat stat, int exec, int visited)
 {
-  UInt line;
-  int nameid;
+    UInt line;
+    int  nameid;
 
-  CheckLeaveFunctionsAfterLongjmp();
+    // Explicitly skip these two cases, as they are often specially handled
+    // and also aren't really interesting statements (something else will
+    // be executed whenever they are).
+    if (TNUM_STAT(stat) == T_TRUE_EXPR || TNUM_STAT(stat) == T_FALSE_EXPR) {
+        return;
+    }
 
-  Int8 ticks = 0, newticks = 0;
+    CheckLeaveFunctionsAfterLongjmp();
 
-  // Explicitly skip these two cases, as they are often specially handled
-  // and also aren't really interesting statements (something else will
-  // be executed whenever they are).
-  if (TNUM_STAT(stat) == T_TRUE_EXPR || TNUM_STAT(stat) == T_FALSE_EXPR) {
-    return;
-  }
+    // Catch the case we arrive here and profiling is already disabled
+    if (!profileState_Active) {
```

Comment:
Off-topic: `profileState_Active` is not used outside this file anymore, as far as I can tell, so it could be moved back into `struct ProfileState`, couldn't it? (I am not suggesting you should do anything about this in this PR, just wondering while I am reading through this code again).
Also, there is no real reason why `fopenMaybeCompressed` and `fcloseMaybeCompressed` take `struct ProfileState* ps` as argument, instead of just accessing the `profileState` global directly, is there?

Reply:
The taking as an argument was beginning progress towards allowing multiple profilestates, so one could profile multiple threads in HPC-GAP, but it was never remotely finished. Should have a cleanup in this area, I agree.